src/types/strings/constrained.rs

```diff
@@ -10,6 +10,7 @@ pub(crate) trait StaticPermittedAlphabet: Sized + Default {
     const CHARACTER_SET: &'static [u32];
     const CHARACTER_WIDTH: u32 = crate::num::log2(Self::CHARACTER_SET.len() as i128);
 
+    fn alphabet_name() -> &'static str;
```

Comment:
I think this would be better as a enum, and have it as a associated const instead of a fn.